### ISSUE TITLE
How to define own loss function, tks

### ISSUE THREAD

**Issue by lidongxing** [tensorflow/tensor2tensor#1734]
### Description

...

### Environment information

```
OS: <your answer here>

$ pip freeze | grep tensor
# your output here

$ python -V
# your output here
```

### For bugs: reproduction and error logs

```
# Steps to reproduce:
...
```

```
# Error logs:
...
```
